substrate/pkg/controller/substrate/cluster/instanceprofile.go

```diff
@@ -31,7 +31,25 @@ type InstanceProfile struct {
 	IAM *iam.IAM
 }
 
-var AssumeRolePolicyDocument = aws.String(`{
+type roleInfo struct {
```

Comment:
Describing a variable name with _Info should trigger naming alarm bells in your head. What value does `Info` convey here? I'd probably just name this `role`, since this concept encapsulates a role in this package.